### PR TITLE
Implement maxValueExclusive and minValueExclusive

### PR DESCRIPTION
**Pull request by bterlson**
Also cleaned up a few surrounding issues I found in the documentation.

### CODE REVIEW

File: packages/compiler/lib/decorators.ts [hunk 1]
```diff
@@ -419,6 +432,85 @@ export function getMaxValue(program: Program, target: Type): number | undefined
...
+    return;
+  }
+  program.stateMap(minValueExclusiveKey).set(target, minValueExclusive);
+}
+
+export function getMinValueExclusive(program: Program, target: Type): number | undefined {
+  return program.stateMap(minValueExclusiveKey).get(target);
+}
+
+// -- @maxValue decorator ---------------------
```
Comment (nguerrera) [inline packages/compiler/lib/decorators.ts:475]:
```suggestion
// -- @maxValueExclusive decorator ---------------------
```

**Review by nguerrera — APPROVED** [review]

File: packages/compiler/lib/decorators.ts [hunk 2]
```diff
@@ -419,6 +432,85 @@ export function getMaxValue(program: Program, target: Type): number | undefined
...
+
+const minValueExclusiveKey = createStateSymbol("minValueExslusive");
+
+export function $minValueExclusive(
+  context: DecoratorContext,
+  target: Scalar | ModelProperty,
+  minValueExclusive: number
+) {
+  validateDecoratorUniqueOnNode(context, target, $minValueExclusive);
+  validateDecoratorNotOnNode(context, target, $minValue, $minValueExclusive);
```
Comment (timotheeguerin) [inline packages/compiler/lib/decorators.ts:445]: What happens if you have 
```
@maxValue(10)
scalar A extends float32;

@maxValueExclusive(10)
scalar A extends B;
```

File: packages/compiler/lib/decorators.ts [hunk 3]
```diff
@@ -419,6 +432,85 @@ export function getMaxValue(program: Program, target: Type): number | undefined
...
+      getMinValue(context.program, target) ?? getMinValueExclusive(context.program, target),
+      maxValueExclusive
+    )
+  ) {
+    return;
+  }
+  program.stateMap(maxValueExclusiveKey).set(target, maxValueExclusive);
+}
+
+export function getMaxValueExclusive(program: Program, target: Type): number | undefined {
```
Comment (timotheeguerin) [inline packages/compiler/lib/decorators.ts:510]: Do we need new tests for those decorators?

Comment (bterlson) [reply packages/compiler/lib/decorators.ts:510]: Yep just added them.

File: packages/compiler/lib/decorators.ts (hunk 2 above)
Comment (bterlson) [reply packages/compiler/lib/decorators.ts:445]: Good question! Right now, not an error. But I believe this should be an error. I will implement that.

File: packages/compiler/core/decorator-utils.ts [hunk 4]
```diff
@@ -432,14 +432,32 @@ export function validateDecoratorUniqueOnNode(
...
+ * Validate that a given decorator is not on a type or any of its base types.
+ * Useful to check for decorator usage that conflicts with another decorator.
+ * @param context Decorator context
+ * @param type The type to check
+ * @param badDecorator The decorator we don't want present
+ * @param givenDecorator The decorator that is the reason why we don't want the bad decorator present
+ * @param includeHeritage Whether to check base types for the bad decorator too
+ * @returns Whether the decorator application is valid
+ */
 export function validateDecoratorNotOnNode(
```
Comment (timotheeguerin) [inline packages/compiler/core/decorator-utils.ts:445]: I guess if this method is the one checking for the heritage does it still make sense to be called like that?
The goal of the other one validateDecoratorOnNode was to emit an error if the same decortor was used twice on teh same node but would not complain if you used it twice via `op is` or `model is`

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**